src/commissaire_http/handlers/clusters.py

```diff
@@ -25,11 +29,222 @@ def list_clusters(message, bus):
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        old_hosts = set(message['params']['old'])  # Ensures no duplicates
+        new_hosts = set(message['params']['new'])  # Ensures no duplicates
+        LOGGER.debug('old_hosts="{}", new_hosts="{}"'.format(
+            old_hosts, new_hosts))
+    except Exception as error:
+        return return_error(message, error, JSONRPC_ERRORS['BAD_REQUEST'])
+
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+    except Exception as error:
+        return return_error(message, error, JSONRPC_ERRORS['NOT_FOUND'])
+
+    if old_hosts != set(cluster['result']['hostset']):
+        msg = 'Conflict setting hosts for cluster {0}'.format(
+            cluster['result']['name'])
+        LOGGER.error(msg)
+        return return_error(message, msg, JSONRPC_ERRORS['CONFLICT'])
+
+    cluster['result']['hostset'] = list(new_hosts)
+    cluster = models.Cluster.new(**cluster['result'])
+    cluster._validate()
+    response = bus.request(
+        'storage.save', params=[
+            'Cluster', cluster.to_dict()])
+    return create_response(message['id'], response['result'])
+
+
+def check_cluster_member(message, bus):
+    """
+    Checks is a member is part of the cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+        if message['params']['host'] in cluster['result']['hostset']:
+            # Return back the host in a list
+            return create_response(message['id'], [message['params']['host']])
+        else:
+            return create_response(
+                message['id'],
+                error='The requested host is not part of the cluster.',
+                error_code=JSONRPC_ERRORS['NOT_FOUND'])
+    except Exception as error:
+        return create_response(
+            message['id'],
+            error=error,
+            error_code=JSONRPC_ERRORS['INTERNAL_ERROR'])
+
+
+def add_cluster_member(message, bus):
+    """
+    Adds a member to the cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+        if message['params']['host'] in cluster['result']['hostset']:
+            # Return back the host in a list ... it's already there
+            return create_response(message['id'], [message['params']['host']])
```

Comment:
nit: Same `return` statement below.  Slightly clearer to invert the condition and keep the `else` branch:
```
if message['params']['host'] not in cluster['result']['hostset']:
    cluster['result']['hostset'].append(message['params']['host'])
    ...

return create_response(message['id'], [message['params']['host']])
```